### DISCUSSION TITLE
Retrieve content from Yaml Single File Collection

### DISCUSSION THREAD

**Discussion by matbrgz** [11ty/eleventy#1943] [Q&A]
I'm converting a Jekyll blog to E11y, but I have the following collection (one file collection):

src/_data/navigation.yml:

---
items:I
  - name: "HOME"
    link: /
    subitems:
      - name: "Quem sou"
        link: /#quemsou
      - name: "Curriculum Vitae"
        link: /#CV
      - name: "Áreas de Conhecimento"
        link: /#areas-de-conhecimento
  - name: "Portfolio"
    link: /portfolio
  - name: "Fale Comigo"
    link: /fale-comigo
  - name: "Blog"
    link: /blog
src/_includes/navigation.html:

{% assign navurl = url | remove: 'index.html' %}
<ul>
  {% for item in collections.navigation %}
  <li{% if item.data.subitems %} class="subnav" {% endif %}>
    <a href="{{ site.baseurl }}{{ item.data.link }}" {% if item.data.url==navurl %} class="curpage" data-current="current page" {%
      endif %}>{{ item.fileSlug }}</a>
    {% if item.data.subitems %}
    {% include "navigation.html", navigation: "item.data.subitems" %}
    {% endif %}
    </li>
    {% endfor %}
</ul>
I'm getting the error:
Cannot read property 'split' of undefined, file:/home/breguez/Dev/matbrgz.com/src/_includes/navigation.html, line:1

My .eleventy.js:

const pkg = require("./package.json").eleventyConfig;

[...]

eleventyConfig.addCollection("navigation", function (collectionApi) {
  return collectionApi.getFilteredByGlob("_data/navigation.yml");
});

return {
  dir: pkg.dir
};

package.json

[...]
"eleventyConfig": {
    "dir": {
      "input": "src",
      "output": "dist",
      "includes": "_includes",
      "layouts": "_layouts",
      "data": "globals"
    }
  },
[...]
How to retrieve content from one Yaml file?

**Comment by pdehaan** [comment]
What was your solution? I managed to get it working w/ a global JSON file and didn't need a custom collection.
https://github.com/pdehaan/11ty-1943
